server/belga/search_providers.py

```diff
@@ -361,7 +362,136 @@ def format_list_item(self, data):
         }
 
 
+class BelgaPressSearchProvider(superdesk.SearchProvider):
+
+    GUID_PREFIX = 'urn:belga.be:belgapress:'
+
+    label = 'Belga Press'
+    base_url = 'https://bp-api.ssl.belga.be/belgapress/api'
+    search_endpoint = 'newsobjects'
+    items_field = 'data'
+    openid_provider_url = 'https://sso.ssl.belga.be/auth/realms/belga/protocol/openid-connect/token'
+    PERIODS = {
+        'day': {'days': 0},
+        'yesterday': {'days': -1},
+        'this-week': {'weekday': 0, 'days': -7},
+        'week': {'weeks': -1},
+        'month': {'months': -1},
+        'year': {'years': -1},
+    }
+
+    def __init__(self, provider: Dict[str, Dict], **kwargs):
+        super().__init__(provider, **kwargs)
+        self.session = requests.Session()
+        self._access_token = None
+        config = provider.get('config', {})
+        if config.get('username') and config.get('password'):
+            self.auth()
+
+    def auth(self):
+        resp = self.session.post(f'{self.openid_provider_url}?scope=openid%20profile', data={
+            'client_id': self.provider.get('config', {}).get('username'),
+            'client_secret': self.provider.get('config', {}).get('password'),
+            'grant_type': 'client_credentials'
+        })
+        resp.raise_for_status()
+        if resp.status_code == 200 and resp.content:
+            data = resp.json()
+            self._access_token = data.get('access_token')
+
+    def find(self, query: Dict[str, Any], params: Optional[Dict] = None):
```

Comment:
It would be nice to have a typing module (`superdesk.typing` or `superdesk.typing.search_provider`) which declare [precise type](https://mypy.readthedocs.io/en/stable/more_types.html#typeddict) for common dict object like `query`, `params`